eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRangeQueriesSpec.scala

```diff
@@ -127,4 +131,226 @@ class ChannelRangeQueriesSpec extends FunSuite {
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, false) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2))
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, true) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2 | INCLUDE_NODE_ANNOUNCEMENT_1 | INCLUDE_NODE_ANNOUNCEMENT_2))
   }
+
+  def makeShortChannelIds(height: Int, count: Int): List[ShortChannelId] = {
+    val output = ArrayBuffer.empty[ShortChannelId]
+    var txIndex = 0
+    var outputIndex = 0
+    while (output.size < count) {
+      if (Random.nextBoolean()) {
+        txIndex = txIndex + 1
+        outputIndex = 0
+      } else {
+        outputIndex = outputIndex + 1
+      }
+      output += ShortChannelId(height, txIndex, outputIndex)
+    }
+    output.toList
+  }
+
+  def validate(chunk: ShortChannelIdsChunk) = {
+    require(!chunk.shortChannelIds.exists(!Router.keep(chunk.firstBlock, chunk.numBlocks, _)))
```

Comment:
nit: a bit more readable imo:

```suggestion
    require(chunk.shortChannelIds.forall(Router.keep(chunk.firstBlock, chunk.numBlocks, _)))
```